### PR TITLE
Dashboard friends

### LINKED ISSUE #11: Dashboard: Friends
As an authenticated user,
I see a section for friends,
In this section, there should be a text field to enter a friend's email and a button to "Add Friend"

**Scenarios**: 
1. If I have not added any friends there should be a message. "You currently have no friends".

2. If I have added friends, I should see a list of all my friends.

**Details:** Users should be able to add a friend by their email address, as long as, the friend is a user of our application and exists in our database. 

**Example:** 
Bugs Bunny and Lola Bunny are users of our application, but Daffy Duck is not. 
1. When Bugs Bunny enters `lola_bunny@gmail.com` to add friend it will be successful and Lola should show up as Bugs Bunny's friend.
2. When Bugs Bunny enters `daffy_duck@gmail.com` to add friend it should give an error message that the user does not exist. 

**Tips:** You'll want to research self-referential `has_many through`. [Here](http://blog.hasmanythrough.com/2007/10/30/self-referential-has-many-through) is a good starting point to understand the concept. You will probably need to do more googling but that's part of the fun ;)

- [ ] Write a happy path test 
- [ ] Write a sad path test

### PR DESCRIPTION
**Pull request by priyapower**
**Contributors:** Priya

Closes #11 

**Details:** (what does this pull request accomplish?)
- Friends
    - Can add a friend using the form in the friends section
    - Error if friend not in database
    - Updates discover header for dynamic name
    - Can see friends I currently have
    - If I add you, you automatically add me back
- Database Updates
    - users should have a name
    - friendships joins updated to step away from follower followee and just have a basic self-referential definition "friend" that is actually an instance of class "user"
    - parties re-added with better name ability on columns
    - party participants re-added with user references instead of the accidental friendship join from before

**Improvements:** (issues/bugs/refactoring?)
- Needs soooo much help in the views
    - There are comments for wireframe matching
- **Sad Test: what if I try to add a friend who I have already added as a friend?**

- SimpleCov (%): 114 / 129 LOC (88.37%)
- SimpleCov Model (%): 31 / 163 LOC (19.02%)
- SimpleCov Feature (%): 103 / 131 LOC (78.63%)
- RuboCop (Complaints/Violations): 16 files inspected, 16 offenses detected

### CODE REVIEW

File: app/controllers/discover_controller.rb [hunk 1]
```diff
@@ -1,3 +1,6 @@
 class DiscoverController < ApplicationController
-  def show; end
+  before_action :require_user
+  def show
+    @user = current_user
+  end
```
Comment (Arique1104) [inline app/controllers/discover_controller.rb:5]: Nice update!

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -15,11 +15,30 @@ def create
...
+      new_friend.friendships.create!(friend:current_user)
+    end
+    redirect_to "/dashboard"
   end
 
   private
 
   def user_params
-    params.permit(:email, :password, :password_confirmation)
+    params.permit(:name, :email, :password, :password_confirmation)
```
Comment (Arique1104) [inline app/controllers/users_controller.rb:42]: I like the name collection, I think it'll give users a nice personal touch.

File: spec/features/dashboard/dashboard_spec.rb [hunk 3]
```diff
@@ -3,35 +3,32 @@
...
-      # - This MAY have to be top down so that the features can drive the database
-      # - We may have to go back to the idea of an array of participant ids (this seems easier to update in the long run)
-      # binding.pry
+      @twilight_sparkle.friendships.create!(friend:@spike)
+      @spike.friendships.create!(friend:@twilight_sparkle)
+      @twilight_sparkle.friendships.create!(friend:@starlight_glimmer)
+      @starlight_glimmer.friendships.create!(friend:@twilight_sparkle)
+      @rainbow_dash.friendships.create!(friend:@twilight_sparkle)
+      @twilight_sparkle.friendships.create!(friend:@rainbow_dash)
+      # @party_1  = @twilight_sparkle.parties.create!(movie_id:"4", runtime:"12", date:"Oct 12, 1940", start:"12 pm")
```
Comment (Arique1104) [inline spec/features/dashboard/dashboard_spec.rb:23]: I'm now realizing that we don't have a run time for our API.

**Review by Arique1104 — APPROVED** [review]
Looks great Priya!

File: spec/features/dashboard/dashboard_spec.rb (hunk 3 above)
Comment (priyapower) [reply spec/features/dashboard/dashboard_spec.rb:23]: Love that you caught this! Let's work on the Movie Details section together so we can discover the solution together!

**Pull request merged**